setuptools/_normalization.py

```diff
@@ -65,7 +65,10 @@ def safe_version(version: str) -> str:
         return str(packaging.version.Version(attempt))
 
 
-def best_effort_version(version: str) -> str:
+def best_effort_version(
+    version: str,
+    template: str = "{safe}.dev0+sanitized.{sanitized}",
```

Comment:
This customisable template parameter is not strictly necessary, it can be removed if we don't mind versions looking like `78.1.0.dev0+sanitized.2.g3a3144f0d`